PR: https://github.com/odoo/odoo/pull/55222

From: 6fd1855f37cf5e221f8d11e276135d0ac705bbb7
From: Andrea Grazioso (agr-odoo)
Date: 2020-07-30 15:43:33

Structural Changes: 3
Total Changes: 6

[FIX] hr_recruitment: allow users to create applications

After 9ddfc18b4132daa395c4fa1591b1c001b8fb2628
an user with recruitment/officer (user) permission on hr_recruitment
could not create applications anymore because doing so imply a write to
hr.recruitment.stage

opw-2305646

closes odoo/odoo#55222

X-original-commit: b8359c55e6e57035ffa3b10bfa5294e22a74e837
Signed-off-by: Nicolas Martinelli (nim) <nim@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/hr_recruitment.py
+++ b/addons/hr_recruitment/models/hr_recruitment.py
@@ -149,9 +149,9 @@ class Applicant(models.Model):
         ('done', 'Green'),
         ('blocked', 'Red')], string='Kanban State',
         copy=False, default='normal', required=True)
-    legend_blocked = fields.Char(related='stage_id.legend_blocked', string='Kanban Blocked', readonly=False)
-    legend_done = fields.Char(related='stage_id.legend_done', string='Kanban Valid', readonly=False)
-    legend_normal = fields.Char(related='stage_id.legend_normal', string='Kanban Ongoing', readonly=False)
+    legend_blocked = fields.Char(related='stage_id.legend_blocked', string='Kanban Blocked', readonly=False, related_sudo=True)
+    legend_done = fields.Char(related='stage_id.legend_done', string='Kanban Valid', readonly=False, related_sudo=True)
+    legend_normal = fields.Char(related='stage_id.legend_normal', string='Kanban Ongoing', readonly=False, related_sudo=True)
     application_count = fields.Integer(compute='_compute_application_count', help='Applications with the same email')
     meeting_count = fields.Integer(compute='_compute_meeting_count', help='Meeting Count')
     refuse_reason_id = fields.Many2one('hr.applicant.refuse.reason', string='Refuse Reason', tracking=True)
